### PR TITLE
feat(speech): adds v2 adaptation samples

### PR DESCRIPTION
**Pull request by telpirion**
Canonical samples are [here](https://github.com/googleapis/python-speech/pull/460).

### CODE REVIEW

**Review by yil532 — APPROVED** [review]
LGTM

File: speech/system-test/adaptationCustomClassReference.v2.test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+const {describe, it} = require('mocha');
+const sinon = require('sinon');
+const uuid = require('uuid');
+const speech = require('@google-cloud/speech').v2;
+
+const {createCustomClassV2} = require('../adaptationCustomClassReference.v2');
+
+const text = 'how old is the Brooklyn Bridge';
+
+let recognizerName, projectId, customClassName, phraseSetName;
```
Comment (ace-n) [inline speech/system-test/adaptationCustomClassReference.v2.test.js:27]: Nit: can these be defined as `const`s here?

File: speech/system-test/adaptationInlineCustomClass.v2.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+const {describe, it} = require('mocha');
+const sinon = require('sinon');
+const uuid = require('uuid');
+const speech = require('@google-cloud/speech').v2;
+
+const {buildInlineCustomClassV2} = require('../adaptationInlineCustomClass.v2');
+
+const text = 'how old is the Brooklyn Bridge';
+
+let recognizerName, projectId;
```
Comment (ace-n) [inline speech/system-test/adaptationInlineCustomClass.v2.test.js:27]: Nit: defined as `const`s here?

File: speech/system-test/adaptationInlineCustomClass.v2.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      recognizerId: `rec-${uuid.v4()}`,
+      recognizer: {
+        languageCodes: ['en-US'],
+        model: 'latest_short',
+      },
+    };
+
+    const operation = await client.createRecognizer(recognizerRequest);
+    const recognizer = operation[0].result;
+    recognizerName = recognizer.name;
```
Comment (ace-n) [inline speech/system-test/adaptationInlineCustomClass.v2.test.js:55]: If this value is deterministic, we should compute it within the test itself.

(If this is **not** deterministic, this line is OK as-is.)

File: speech/system-test/adaptationInlineCustomClass.v2.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  beforeEach(stubConsole);
+  afterEach(restoreConsole);
+
+  it('should create an inline class reference and phrase set', async () => {
+    await buildInlineCustomClassV2(recognizerName);
+    expect(console.log.firstCall.args[0]).to.contain(text);
```
Comment (ace-n) [inline speech/system-test/adaptationInlineCustomClass.v2.test.js:63]: We previously decided to move off of `expect` syntax to the [Node.js `assert` API](https://nodejs.org/api/assert.html).

_For consistency_, can we use the `assert` API here instead?

**Review by ace-n — CHANGES_REQUESTED** [review]

File: speech/system-test/adaptationCustomClassReference.v2.test.js (hunk 1 above)
Comment (telpirion) [reply speech/system-test/adaptationCustomClassReference.v2.test.js:27]: No, they're not deterministic as the names will depend on the projectId.

File: speech/system-test/adaptationInlineCustomClass.v2.test.js (hunk 2 above)
Comment (telpirion) [reply speech/system-test/adaptationInlineCustomClass.v2.test.js:27]: No, they're not deterministic as the names will depend on the projectId.

File: speech/system-test/adaptationInlineCustomClass.v2.test.js (hunk 3 above)
Comment (telpirion) [reply speech/system-test/adaptationInlineCustomClass.v2.test.js:55]: No, they're not deterministic as the names will depend on the projectId.

File: speech/system-test/adaptationInlineCustomClass.v2.test.js (hunk 4 above)
Comment (telpirion) [reply speech/system-test/adaptationInlineCustomClass.v2.test.js]: Done.

**Review by ace-n — APPROVED** [review]

**Pull request merged**